src/voting.ts

```diff
@@ -117,23 +89,52 @@ export function handleSupply(event: Supply): void {
 }
 
 export function handleUserBalFromChain(event: UserBalFromChain): void {
-  setOmniVotingEscrowLock(
-    event.address,
-    event.params.user,
-    'From',
-    event.params.srcChainId,
-    event.params.userPoint.bias,
-    event.params.userPoint.slope,
-  );
+  let userAddress = event.params.user;
+  createUserEntity(userAddress);
+
+  let id = getVotingEscrowId(userAddress, event.address);
+  let votingShare = VotingEscrowLock.load(id);
+
+  if (votingShare == null) {
+    votingShare = new VotingEscrowLock(id);
+    votingShare.user = userAddress.toHexString();
+    votingShare.votingEscrowID = event.address.toHexString();
+  }
+
+  // TODO: is event.params.ts == blockTimestamp?
+  votingShare.unlockTime = event.params.userPoint.ts;
+  votingShare.updatedAt = event.block.timestamp.toI32();
+
+  const lockedBalanceBI = event.params.userPoint.slope.times(LOCK_MAXTIME);
+  votingShare.lockedBalance = scaleDownBPT(lockedBalanceBI);
+
+  votingShare.slope = scaleDownBPT(event.params.userPoint.slope);
+  votingShare.bias = scaleDownBPT(event.params.userPoint.bias);
+
+  votingShare.save();
 }
 
 export function handleUserBalToChain(event: UserBalToChain): void {
-  setOmniVotingEscrowLock(
-    event.address,
-    event.params.localUser,
-    'To',
-    event.params.dstChainId,
-    event.params.userPoint.bias,
-    event.params.userPoint.slope,
-  );
+  // TODO: find diff between localUser and remoteUser
+  // maybe create 2 separate fields for this
```

Comment:
Yes, agree we should store `localUser` and `remoteUser` in the `OmniVotingEscrowLock` entity

Reply:
I made only `localUser` a reference to `User` entity and left `remoteUser` as a bytes field - wdyt? https://github.com/balancer/gauges-subgraph/pull/34/commits/183c3b8e567c5e77ff125b91107b44dcb4c60bbc